mapviz_plugins/src/pointcloud2_plugin.cpp

```diff
@@ -654,28 +681,35 @@ namespace mapviz_plugins
 
   void PointCloud2Plugin::Transform()
   {
-    std::deque<Scan>::iterator scan_it = scans_.begin();
-    for (; scan_it != scans_.end(); ++scan_it)
     {
-      Scan& scan = *scan_it;
+      QMutexLocker locker(&scan_mutex_);
 
-      swri_transform_util::Transform transform;
+      std::deque<Scan>::iterator scan_it = scans_.begin();
       bool was_using_latest_transforms = use_latest_transforms_;
       use_latest_transforms_ = false;
-      if (GetTransform(scan.source_frame, scan.stamp, transform))
+      for (; scan_it != scans_.end(); ++scan_it)
       {
-        scan.transformed = true;
-        std::vector<StampedPoint>::iterator point_it = scan.points.begin();
-        for (; point_it != scan.points.end(); ++point_it)
+        Scan& scan = *scan_it;
+
+        if (!scan_it->transformed)
```

Comment:
Does caching the transformed points have any possible nasty side effects? I'm asking because I honestly don't know, and caching makes me nervous.